File: app/src/main/java/cn/tongdun/android/adapter/DeviceInfoItemAdapter.java
package cn.tongdun.android.adapter;

import android.text.TextUtils;
import android.util.Pair;
import android.view.View;
import android.widget.TextView;

import com.trustdevice.android.R;

import java.util.List;

/**
 * @description: BuildInfo Item Adapter
 * @author: wuzuchang
 * @date: 2022/12/6
 */
public class DeviceInfoItemAdapter extends BaseRecyclerViewAdapter<Pair<String, String>> {

    public DeviceInfoItemAdapter(List<Pair<String, String>> data) {
        super(data);
    }

    @Override
    public int getLayoutId(int viewType) {
        return R.layout.item_details;
    }

    @Override
    public void convert(BaseViewHolder holder, Pair<String, String> data, int position) {
        TextView name = holder.getView(R.id.tv_name);
        name.setText(data.first);
        TextView value = holder.getView(R.id.tv_value);
        if (TextUtils.isEmpty(data.second)) {
            value.setVisibility(View.GONE);
        } else {
            value.setText(data.second);
        }
    }
}

File: app/src/main/kotlin/cn/tongdun/android/adapters/DeviceInfoItemAdapter.kt
package cn.tongdun.android.adapters

import android.text.TextUtils
import android.util.Pair
import android.view.View
import android.widget.TextView
import com.trustdevice.android.R


class DeviceInfoItemAdapter(data: List<Pair<String, String>>) : BaseRecyclerViewAdapter<Pair<String, String>>(data) {
    override fun getLayoutId(viewType: Int): Int {
        return R.layout.item_details
    }

    override fun convert(holder: BaseViewHolder, data: Pair<String, String>, position: Int) {
        val name = holder.getView<TextView>(R.id.tv_name)
        name?.text = data.first
        val value = holder.getView<TextView>(R.id.tv_value)
        if (TextUtils.isEmpty(data.second)) {
            value?.visibility = View.GONE
        } else {
            value?.text = data.second
        }
    }
}
